extension/src/popup/views/VerifyAccount/index.tsx

```diff
@@ -32,70 +28,49 @@ export const VerifyAccount = ({
   const dispatch = useDispatch();
-  const authError = useSelector(authErrorSelector);
+
+  const publicKey = useSelector(publicKeySelector);
+
   const from = get(location, "state.from.pathname", "") as ROUTES;
 
-  const handleSubmit = async (values: { password: string }) => {
+  const history = useHistory();
+
+  const handleConfirm = async (password: string) => {
     if (customSubmit) {
-      await customSubmit(values.password);
+      await customSubmit(password);
     } else {
       // eslint-disable-next-line
-      await dispatch(confirmPassword(values.password));
+      await dispatch(confirmPassword(password));
       navigateTo(from || ROUTES.account);
     }
   };
 
+  const handleCancel = () => {
+    if (customBackAction) {
+      customBackAction();
+      return;
+    }
+
+    history.goBack();
+  };
+
   return (
     <React.Fragment>
-      <SubviewHeader
-        title={t("Verification")}
-        customBackAction={customBackAction}
-      />
-      <Formik initialValues={{ password: "" }} onSubmit={handleSubmit}>
-        {({ dirty, isValid, isSubmitting, errors, touched }) => (
-          <Form className="View__contentAndFooterWrapper">
-            <View.Content>
-              <div className="VerifyAccount__copy">
-                {isApproval
-                  ? t("Enter your account password to verify your account.")
-                  : t(
-                      "Enter your account password to authorize this transaction.",
-                    )}{" "}
-                You won’t be asked to do this for the next 24 hours.
-              </div>
-
-              <Field name="password">
-                {({ field }: FieldProps) => (
-                  <Input
-                    fieldSize="md"
-                    id="password-input"
-                    type="password"
-                    placeholder={t("Enter password")}
-                    error={
-                      authError ||
-                      (errors.password && touched.password
-                        ? errors.password
-                        : "")
-                    }
-                    {...field}
-                  />
-                )}
-              </Field>
-            </View.Content>
-            <View.Footer>
-              <Button
-                size="md"
-                isFullWidth
-                variant="primary"
-                isLoading={isSubmitting}
-                disabled={!(dirty && isValid)}
-              >
-                {isApproval ? t("Approve") : t("Submit")}
-              </Button>
-            </View.Footer>
-          </Form>
-        )}
-      </Formik>
+      <div className="VerifyAccount">
+        <EnterPassword
+          accountAddress={publicKey}
+          description={
+            isApproval
+              ? undefined
+              : t(
+                  "Enter your account password to authorize this transaction. You won’t be asked to do this for the next 24 hours.",
+                )
```

Comment:
NIT: I think `You won’t be asked to do this for the next 24 hours.` should be part of the `EnterPassword` component since it's used by both the default `description` and this custom one.

The only reason I even bring it up is because we're hopefully going to make the amount of time configurable soon so it'd be nice to only have to deal with this copy in one place

Reply:
Good point. I was in the fence about this too. Do you think we should also display this 24h warning on the login screen below?

![Screenshot 2024-12-05 at 09 58 25](https://github.com/user-attachments/assets/6767fbaf-d1ab-4591-b73d-c3a50cc0417d)

Reply:
Oh yeah, that's a good idea!

Reply:
Here we go: https://github.com/stellar/freighter/pull/1706/commits/1ceebe6af730ea04a5952b473835382131939091
![Screenshot 2024-12-05 at 15 36 14](https://github.com/user-attachments/assets/b565d3f5-4c75-4f7b-82b2-8c64fb9d2703)